src/modules/roc_fec/target_openfec/roc_fec/of_decoder.cpp

```diff
@@ -88,13 +71,35 @@ bool OFDecoder::valid() const {
     return valid_;
 }
 
+bool OFDecoder::begin(size_t sblen, size_t rblen) {
+    roc_panic_if_not(valid());
+
+    if (sblen_ == sblen && rblen_ == rblen) {
+        return true;
```

Comment:
No. In most cases when we do not change a block size the workflow will be:

```
decoder.begin() 
   -> decoder.set() 
   -> decoder.end()  // here we will reset a session
```

Most of the time we will call `begin` with the same parameters and this `if` is some kind of optimization to avoid recreating FEC session params each time.

Reply:
According to https://github.com/roc-project/roc/pull/174#discussion_r277382695 I removed this if block.